slack-java-client/src/main/java/com/hubspot/slack/client/SlackClientFactory.java

```diff
@@ -0,0 +1,42 @@
+package com.hubspot.slack.client;
+
+import com.hubspot.slack.client.http.NioHttpClientFactory;
+
+public interface SlackClientFactory extends SlackWebClient.Factory {
+
+  static SlackClientFactory defaultFactory() {
+    return DefaultSlackClientFactory.defaultFactory();
+  }
+
+  SlackClient create(SlackClientRuntimeConfig config);
+
+  SlackClient create(NioHttpClientFactory httpClientFactory, SlackClientRuntimeConfig config);
+
+  static class DefaultSlackClientFactory implements SlackClientFactory {
+    private static final SlackClientFactory INSTANCE = new DefaultSlackClientFactory();
+
+    public static SlackClientFactory defaultFactory() {
+      return INSTANCE;
+    }
+
+    /**
+     * Use {@link SlackClientFactory#create(SlackClientRuntimeConfig)} instead
+     */
+    @Override
+    @Deprecated
+    public SlackWebClient build(SlackClientRuntimeConfig config) {
+      return new SlackWebClient(NioHttpClientFactory.defaultFactory(), config);
+    }
```

Comment:
Is there a reason for this method to exist? Since this is new code can we just make everyone call `create` instead?

Reply:
The only reason it exists is to have one implementation of the `SlackClientFactory`, which extends the existing `@AssistedIInject` interface.  Keeping this allows us to bind only these implementations without worrying that we will break code that hasn't migrated yet

Reply:
Ah ok I missed the `@Override` at first. In the future will that stuff move to the Guice module? Or just get replaced by something else?

Reply:
The plan is to release this code, migrate our internal version to this and start moving people to the new methods and the new `slack-java-client-guice` module.  Then we can delete the `@AssistedInject` factory interface in the `SlackWebClient`, remove this method and drop the slack client module from `slack-java-client` along with all guice-related dependencies

Reply:
Sounds good 👍